include/zephyr/bluetooth/services/bas_bls.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2024 Demant A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_BAS_BLS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_BAS_BLS_H_
+
+#include <sys/types.h>
+#include <zephyr/sys/atomic.h>
+#include <stdint.h>
+#include <zephyr/types.h>
+
+/* Enums for battery level status fields */
+typedef enum {
+	BT_BAS_BLS_BATTERY_NOT_PRESENT = 0,
+	BT_BAS_BLS_BATTERY_PRESENT = 1
+} bt_bas_bls_battery_present_t;
+
+typedef enum {
+	BT_BAS_BLS_WIRED_POWER_NOT_CONNECTED = 0,
+	BT_BAS_BLS_WIRED_POWER_CONNECTED = 1,
+	BT_BAS_BLS_WIRED_POWER_UNKNOWN = 2
+} bt_bas_bls_wired_power_source_t;
+
+typedef enum {
+	BT_BAS_BLS_WIRELESS_POWER_NOT_CONNECTED = 0,
+	BT_BAS_BLS_WIRELESS_POWER_CONNECTED = 1,
+	BT_BAS_BLS_WIRELESS_POWER_UNKNOWN = 2
+} bt_bas_bls_wireless_power_source_t;
+
+typedef enum {
+	BT_BAS_BLS_CHARGE_STATE_UNKNOWN = 0,
+	BT_BAS_BLS_CHARGE_STATE_CHARGING = 1,
+	BT_BAS_BLS_CHARGE_STATE_DISCHARGING_ACTIVE = 2,
+	BT_BAS_BLS_CHARGE_STATE_DISCHARGING_INACTIVE = 3
+} bt_bas_bls_battery_charge_state_t;
+
+typedef enum {
+	BT_BAS_BLS_CHARGE_LEVEL_UNKNOWN = 0,
+	BT_BAS_BLS_CHARGE_LEVEL_GOOD = 1,
+	BT_BAS_BLS_CHARGE_LEVEL_LOW = 2,
+	BT_BAS_BLS_CHARGE_LEVEL_CRITICAL = 3
+} bt_bas_bls_battery_charge_level_t;
+
+typedef enum {
+	BT_BAS_BLS_CHARGE_TYPE_UNKNOWN = 0,
+	BT_BAS_BLS_CHARGE_TYPE_CONSTANT_CURRENT = 1,
+	BT_BAS_BLS_CHARGE_TYPE_CONSTANT_VOLTAGE = 2,
+	BT_BAS_BLS_CHARGE_TYPE_TRICKLE = 3,
+	BT_BAS_BLS_CHARGE_TYPE_FLOAT = 4
+} bt_bas_bls_battery_charge_type_t;
+
+typedef enum {
+	BT_BAS_BLS_FAULT_REASON_NONE = 0,
+	BT_BAS_BLS_FAULT_REASON_BATTERY = 1,
+	BT_BAS_BLS_FAULT_REASON_EXTERNAL_POWER = 2,
+	BT_BAS_BLS_FAULT_REASON_OTHER = 4
+} bt_bas_bls_charging_fault_reason_t;
+
+typedef enum {
+	BT_BAS_BLS_SERVICE_REQUIRED_FALSE = 0,
+	BT_BAS_BLS_SERVICE_REQUIRED_TRUE = 1,
+	BT_BAS_BLS_SERVICE_REQUIRED_UNKNOWN = 2
+} bt_bas_bls_service_required_t;
+
+typedef enum {
+	BT_BAS_BLS_BATTERY_FAULT_NO = 0,
+	BT_BAS_BLS_BATTERY_FAULT_YES = 1
+} bt_bas_bls_battery_fault_t;
+
+/* Battery level status structure definition */
+typedef struct {
+	struct {
+		uint8_t identifier: 1;
+		uint8_t battery_level: 1;
+		uint8_t additional_status: 1;
+		uint8_t flags_reserved: 5;
+	} flags;
+
+	union {
+		struct {
+			uint16_t battery: 1;
+			uint16_t wired_external_power_source: 2;
+			uint16_t wireless_external_power_source: 2;
+			uint16_t battery_charge_state: 2;
+			uint16_t battery_charge_level: 2;
+			uint16_t battery_charge_type: 3;
+			uint16_t charging_fault_reason: 3;
+			uint16_t power_state_reserved: 1;
```

Comment:
The actual ordering of these in memory depend on the architecture (or rather its endianness), so you should either not use these helper bitfields or then you need two variants of it depending on the endianness.